### PR TITLE
fix(components/tiles): fix appending tile components with injected dependencies

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2407698](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2407698)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/945/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/a11y-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/ag-grid-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/core-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/popovers-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/tabs-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/945/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/945/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/945/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/945/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/945/playground/)

File: libs/components/core/src/lib/modules/dynamic-component/dynamic-component.service.ts [hunk 1]
```diff
@@ -67,53 +67,60 @@ export class SkyDynamicComponentService {
...
-        if (!options.referenceEl) {
-          throw new Error(
-            '[SkyDynamicComponentService] Could not create a component at location `SkyDynamicComponentLocation.BeforeElement` because a reference element was not provided.'
+    let componentRef: ComponentRef<T>;
+    if (options.viewRef) {
+      componentRef = options.viewRef.createComponent(componentType, {
+        injector: injector,
+      });
+    } else {
+      componentRef = createComponent<T>(componentType, {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/dynamic-component/dynamic-component.service.ts:76]: Before we merge, can we figure out why `createComponent` wasn't working? What do `environmentInjector` and `elementInjector` do?

Comment (Blackbaud-CoreyArcher) [reply libs/components/core/src/lib/modules/dynamic-component/dynamic-component.service.ts:76]: `createComponent` seems not to want to create a new element with the injection scope we are attempting to pass in. `environmentInjector` is essentially a top level scope, and `elementInjector` is an optional localized one, but for some reason passing this injector in doesn't contain a reference to a service provided at the parent component level.

File: libs/components/core/src/lib/modules/dynamic-component/dynamic-component-options.ts [hunk 2]
```diff
@@ -21,6 +21,11 @@ export interface SkyDynamicComponentOptions {
    */
   referenceEl?: HTMLElement;
 
+  /**
+   * The view container reference where the new component should be appended.
+   */
+  viewRef?: ViewContainerRef;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/dynamic-component/dynamic-component-options.ts]: There is something called `ViewRef` in Angular; let's name this `viewContainerRef`.

Comment (Blackbaud-CoreyArcher) [reply libs/components/core/src/lib/modules/dynamic-component/dynamic-component-options.ts]: done

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**